here-naksha-lib-geo/src/jvmMain/kotlin/naksha/geo/ProxyGeoUtil.kt

```diff
@@ -212,14 +212,14 @@ object ProxyGeoUtil {
      */
     @JvmStatic
     fun toJtsGeometry(geometry: SpGeometry): Geometry {
-        return when (geometry) {
-            is SpPoint -> toJtsPoint(geometry)
-            is SpMultiPoint -> toJtsMultiPoint(geometry)
-            is SpLineString -> toJtsLineString(geometry)
-            is SpMultiLineString -> toJtsMultiLineString(geometry)
-            is SpPolygon -> toJtsPolygon(geometry)
-            is SpMultiPolygon -> toJtsMultiPolygon(geometry)
-            is SpGeometryCollection -> toJtsGeometryCollection(geometry)
+        return when (geometry.type) {
+            Point.TYPENAME_POINT -> toJtsPoint(geometry.asPoint())
+            Point.TYPENAME_MULTIPOINT -> toJtsMultiPoint(geometry.asMultiPoint())
+            Point.TYPENAME_LINESTRING -> toJtsLineString(geometry.asLineString())
+            Point.TYPENAME_MULTILINESTRING -> toJtsMultiLineString(geometry.asMultiLineString())
+            Point.TYPENAME_POLYGON -> toJtsPolygon(geometry.asPolygon())
+            Point.TYPENAME_MULTIPOLYGON -> toJtsMultiPolygon(geometry.asMultiPolygon())
+            Point.TYPENAME_GEOMETRYCOLLECTION -> toJtsGeometryCollection(geometry.asGeometryCollection())
             else -> throw IllegalArgumentException("Unknown proxy type ${geometry::class.simpleName}")
```

Comment:
I changed the code to use `is`, because there is no guarantee the `type` is set correctly. For example:
```Kotlin
val raw = `{"coordinates":[1,2]}`
val json = Platform.fromJSON(raw)
val point = Platform.proxy(json, SpPoint::class)
```
However, I see, we basically should somehow fix this, I recommend to fix `type` to be `NotNullProperty` and in the init code, if no type set, detect the type using the `is` construct. Then adjust all constructors of the SP classes to set the correct `type` value when being created.

Reply:
done